src/osd/OSD.h

```diff
@@ -469,10 +470,16 @@ class OSDService {
     sched_scrub_pg.insert(scrub);
     return scrub.sched_time;
   }
-  void unreg_pg_scrub(spg_t pgid, utime_t t) {
+  void unreg_pg_scrub(spg_t pgid) {
     Mutex::Locker l(sched_scrub_lock);
-    size_t removed = sched_scrub_pg.erase(ScrubJob(cct, pgid, t));
-    ceph_assert(removed);
+    set<ScrubJob>::iterator f;
+    for (f = sched_scrub_pg.begin(); f != sched_scrub_pg.end(); ++f) {
```

Comment:
I don't link this O(n) search to unreg_pg_scrub... which can happen quite a bit if I recall.  Instead, can we modify reg_pg_scrub to make the ScrubJob a member of PG, so that it is always available to do the lookup and unreg?